icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/MlBreakEngine.java

```diff
@@ -406,12 +408,12 @@ private int initElementList(CharacterIterator inString, Element[] elementList,
      * is less than 3 digits.
      */
     private String getUnicodeBlock(int ch) {
-        UCharacter.UnicodeBlock block = UCharacter.UnicodeBlock.of(ch);
-        if (block == UCharacter.UnicodeBlock.NO_BLOCK
-                || block == UCharacter.UnicodeBlock.INVALID_CODE) {
-            return String.valueOf((char) INVALID);
+        int blockId = UCharacter.UnicodeBlock.of(ch).getID();
+        if (blockId == UCharacter.UnicodeBlock.NO_BLOCK.getID()
+                || blockId == UCharacter.UnicodeBlock.INVALID_CODE_ID) {
+            return INVALID_STRING;
         } else {
-            return String.format("%03d", block.getID());
+            return String.format("%03d", blockId);
```

Comment:
```suggestion
            return Integer.toString(blockId);
```

Reply:
I could not transform to String directly since it needs zero padding in front if its length is less than 3.
https://github.com/unicode-org/icu/blob/main/icu4c/source/common/mlbe.cpp#L408

Reply:
ok makes sense tnx